flink-connectors/flink-connector-elasticsearch7/src/main/java/org/apache/flink/connector/elasticsearch/sink/ElasticsearchSinkBuilder.java

```diff
@@ -0,0 +1,276 @@
+ * limitations under the License.
+ */
+
+package org.apache.flink.connector.elasticsearch.sink;
+
+import org.apache.flink.annotation.PublicEvolving;
+import org.apache.flink.connector.base.DeliveryGuarantee;
+import org.apache.flink.streaming.connectors.elasticsearch.ElasticsearchSinkBase;
+import org.apache.flink.util.InstantiationUtil;
+
+import org.apache.http.HttpHost;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.apache.flink.util.Preconditions.checkNotNull;
+import static org.apache.flink.util.Preconditions.checkState;
+
+/**
+ * Builder to construct {@link ElasticsearchSink}.
+ *
+ * <p>The following example shows the minimum setup to create a ElasticsearchSink that submits
+ * actions on checkpoint.
+ *
+ * <pre>{@code
+ * Elasticsearch<String> sink = Elasticsearch
+ *     .builder()
+ *     .setHosts(MY_ELASTICSEARCH_HOSTS)
+ *     .setProcessor(MY_ELASTICSEARCH_PROCESSOR)
+ *     .setDeliveryGuarantee(DeliveryGuarantee.AT_LEAST_ONCE)
+ *     .build();
+ * }</pre>
+ *
+ * @param <IN> type of the records converted to Elasticsearch actions
+ */
+@PublicEvolving
+public class ElasticsearchSinkBuilder<IN> {
+
+    private int bulkFlushMaxActions = -1;
+    private int bulkFlushMaxMb = -1;
+    private long bulkFlushInterval = -1;
+    private ElasticsearchSinkBase.FlushBackoffType bulkFlushBackoffType;
+    private int bulkFlushBackoffRetries = -1;
+    private long bulkFlushBackOffDelay = -1;
+    private DeliveryGuarantee deliveryGuarantee = DeliveryGuarantee.NONE;
+    private List<HttpHost> hosts;
+    private ElasticsearchProcessor<? extends IN> processor;
+    private String username;
+    private String password;
+    private String connectionPathPrefix;
+
+    ElasticsearchSinkBuilder() {}
+
+    /**
+     * Sets the hosts where the Elasticsearch cluster nodes are reachable.
+     *
+     * @param hosts http addresses describing the node locations
+     */
+    public ElasticsearchSinkBuilder<IN> setHosts(HttpHost... hosts) {
+        checkNotNull(hosts);
+        checkState(hosts.length > 0, "Hosts cannot be empty.");
+        this.hosts = Arrays.asList(hosts);
+        return this;
+    }
+
+    /**
+     * Sets the processor which is invoked on every record to convert it to Elasticsearch actions.
+     *
+     * @param processor to process records into Elasticsearch actions.
+     * @return {@link ElasticsearchSinkBuilder}
+     */
+    public <T extends IN> ElasticsearchSinkBuilder<T> setProcessor(
+            ElasticsearchProcessor<? extends T> processor) {
+        checkNotNull(processor);
+        checkState(
+                InstantiationUtil.isSerializable(processor),
+                "The elasticsearch processor must be serializable.");
+        final ElasticsearchSinkBuilder<T> self = self();
+        self.processor = processor;
+        return self;
+    }
+
+    /**
+     * Sets the wanted the {@link DeliveryGuarantee}. The default delivery guarantee is {@link
+     * #deliveryGuarantee}.
```

Comment:
missing default or is this on purpose so comments do not get stale?

Reply:
The javadoc refers via the link to a variable named `deliveryGuarantee`  holding the default value :)

Reply:
Good that we have done the same thing in the Kafka docs :)